internal/platform/storage/badger/badger.go

```diff
@@ -156,3 +158,66 @@ func (t Tx) GetKeysWithPrefix(prefix []byte, reverse ...bool) [][]byte {
 
 	return keys
 }
+
+func (t Tx) GetValuesWithPrefix(prefix []byte, reverse ...bool) [][]byte {
+	var values [][]byte
+
+	// Determine the reverse setting based on the optional parameter
+	isReverse := false
+	if len(reverse) > 0 {
+		isReverse = reverse[0]
+	}
+
+	opts := badger.DefaultIteratorOptions
+	opts.PrefetchValues = true
+	opts.Reverse = isReverse
+	opts.PrefetchSize = 100
+	iterator := t.tx.NewIterator(opts)
+	defer iterator.Close()
+
+	// Append 0xff to the prefix for the seek key if in reverse mode
+	var seekPrefix []byte
+	if isReverse {
+		seekPrefix = append([]byte(nil), prefix...)
+		seekPrefix = append(seekPrefix, 0xff)
+	} else {
+		seekPrefix = prefix
+	}
+
+	for iterator.Seek(seekPrefix); iterator.ValidForPrefix(prefix); iterator.Next() {
+		item := iterator.Item()
+		value, err := item.ValueCopy(nil)
+		if err != nil {
+			continue
+		}
+		values = append(values, value)
+	}
+
+	return values
+}
+
+func (t Tx) FilterKeysWithPrefix(prefix []byte, from, to string) [][]byte {
```

Comment:
I don't see who uses this function?

Reply:
Nobody, but I leave it there because I think we will need it in the future